### PR TITLE
DCS-1516 Adding fluent API for building up routes

### PR DESCRIPTION
**Pull request by andrewrlee**

### CODE REVIEW

**Review by GurnankCheema — APPROVED** [review]
LGTM

File: server/routes/bookedtoday/arrivals/searchforexisting/index.ts [hunk 1]
```diff
@@ -14,50 +12,40 @@ import redirectIfDisabledMiddleware from '../../../../middleware/redirectIfDisab
...
+    .post(
+      `${basePath}/possible-records-found`,
+      redirectIfDisabledMiddleware(config.confirmNoIdentifiersEnabled),
+      checkSearchDetailsPresent,
+      validationMiddleware(MatchedRecordSelectionValidation),
+      multipleMatchFoundController.submit()
+    )
+    .get(
+      `${basePath}/record-found`,
+      redirectIfDisabledMiddleware(config.confirmNoIdentifiersEnabled),
```
Comment (andrewrlee) [inline server/routes/bookedtoday/arrivals/searchforexisting/index.ts:43]: think we can remove this feature toggle now as the functionality is all present, just need to check normal feature toggle is in place for all routes

File: server/utils/routeBuilder.ts [hunk 2]
```diff
@@ -0,0 +1,42 @@
+/* eslint-disable max-classes-per-file */
```
Comment (andrewrlee) [inline server/utils/routeBuilder.ts:1]: a little fluent iterface / DSL to allow defining routes and ensuring role protection is explicitly stated and async middleware applied correctly

**Review by connormaglynn — APPROVED** [review]
lgtm

**Pull request merged**